### ISSUE TITLE
Add Haskell checker

### ISSUE THREAD

**Issue by swsnr** [flycheck/flycheck#32]
Add checkers for Haskell.

Check Haskell buffers for syntax errors first, using `ghc`, and afterwards check for style errors with `hslint`.

Depends on #31.

**Comment by boothead** [comment]
@jwiegley Are you still using the some version of this? I'm struggling to get spacemacs flycheck and hlint working together

**Comment by lunaryorn** [comment]
@boothead Please don't dig up old issues two years after the last change.  It'll needlessly notify users who after two years might not even be using Flycheck anymore, and might now have more important things to do than setting up other people's Emacsen.

Instead please open a new issue, ideally in the Spacemacs tracker first.
